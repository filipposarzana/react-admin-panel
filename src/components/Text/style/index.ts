import { ComponentType, ReactNode } from 'react'
import { css, StyledComponent } from 'styled-components'
import { ColorName, colors } from '~/constants/colors'
import { typographyFlavours, TypographyKind } from '~/constants/typography'

export type TypographyProps = {
  children: ReactNode
  colorName?: ColorName
  fontStyle?: 'normal' | 'italic'
  kind?: TypographyKind
}

export type TextType = StyledComponent<ComponentType<TypographyProps>, any, TypographyProps, never>

const getTypographyKind = ({ kind = 'bodyM' }: Pick<TypographyProps, 'kind'>) => {
  const { fontSize, fontWeight, lineHeight } = typographyFlavours[kind]

  return `
    font-size: ${fontSize}px;
    font-weight: ${fontWeight};
    line-height: ${lineHeight}px;
  `
}

export const baseTypographyStyle = css<TypographyProps>`
  ${getTypographyKind}

  color: ${({ colorName }: TypographyProps) => colors[colorName || 'gray900']};
  font-style: ${({ fontStyle = 'normal' }) => fontStyle};
`
