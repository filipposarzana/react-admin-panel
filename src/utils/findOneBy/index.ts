import { User, users } from '~/db'

type Key = Extract<keyof User, 'name' | 'uuid'>

const findOneBy = <K extends Key>(prop: K) => (value: User[K]) =>
  users.find((user) => user[prop].toLowerCase() === value.toLowerCase())

export const findOneByName = findOneBy('name')

export const findOneByUuid = findOneBy('uuid')
