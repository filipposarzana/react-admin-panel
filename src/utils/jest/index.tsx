import { render } from '@testing-library/react'
import { renderHook } from '@testing-library/react-hooks'
import { ReactElement } from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { StyleSheetManager } from 'styled-components'

export { act, fireEvent, waitFor } from '@testing-library/react'

export const renderer = {
  deep: (node: ReactElement) => {
    const r = render(<StyleSheetManager disableVendorPrefixes>{node}</StyleSheetManager>)

    return {
      ...r,
      output: r.container.firstChild,
    }
  },
  hook: <T extends unknown, R extends unknown>(hookFunction: (props: T) => R) => {
    const r = renderHook(hookFunction)

    return r
  },
  shallow: (node: ReactElement) => {
    const r = createRenderer()

    r.render(node)

    return {
      instance: r.getMountedInstance(),
      output: r.getRenderOutput(),
      unmount: r.unmount,
    }
  },
}
