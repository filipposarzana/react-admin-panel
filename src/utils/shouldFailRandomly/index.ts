export const shouldFailRandomly = () => Math.round(Math.random()) === 0
