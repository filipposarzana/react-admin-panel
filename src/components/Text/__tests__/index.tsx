import { renderer } from '~/utils/jest'
import { TypographyProps } from '../style'
import { Text } from '..'

const fixtures: [string, Omit<TypographyProps, 'children'>][] = [
  ['matches snapshot with default props', {}],
  ['matches snapshot with all props', { colorName: 'gray300', kind: 'headingM' }],
]

describe('Text', () => {
  fixtures.forEach(([title, props]) => {
    it(title, () => {
      const { output } = renderer.deep(<Text {...props}>Text</Text>)

      expect(output).toMatchSnapshot()
    })
  })
})
