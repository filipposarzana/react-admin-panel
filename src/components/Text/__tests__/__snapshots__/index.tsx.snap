// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text matches snapshot with all props 1`] = `
.c0 {
  font-size: 34px;
  font-weight: 700;
  line-height: 40px;
  color: rgb(118,118,118);
  font-style: normal;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
}

<span
  class="c0"
  kind="headingM"
>
  Text
</span>
`;

exports[`Text matches snapshot with default props 1`] = `
.c0 {
  font-size: 15px;
  font-weight: 500;
  line-height: 19px;
  color: rgb(46,46,46);
  font-style: normal;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
}

<span
  class="c0"
>
  Text
</span>
`;
